src/N64.c

```diff
@@ -60,4 +60,43 @@ bool n64_read(const uint8_t pin, N64_Report_t* report)
 // N64 Console
 //================================================================================
 
-// TODO
+uint8_t n64_write(const uint8_t pin, N64_Status_t* status, N64_Report_t* report)
+{
+    // 0 = no input/error, 1 = identify, 2 = status (poll),
+    // 3 = read from expansion bus, 4 = write to expansion bus
+    uint8_t ret = 0;
+
+    // Get the port mask and the pointers to the in/out/mode registers
+    uint8_t bitMask = digitalPinToBitMask(pin);
+    uint8_t port = digitalPinToPort(pin);
+    volatile uint8_t* modePort = portModeRegister(port);
+    volatile uint8_t* outPort = portOutputRegister(port);
+    volatile uint8_t* inPort = portInputRegister(port);
+
+    // Don't want interrupts getting in the way
+    uint8_t oldSREG = SREG;
+    cli();
+
+    // Read in data from the console
+    // After receiving the init command you have max 20ms to respond (for the data command)!
+    uint8_t command[3] = {0, 0, 0};
```

Comment:
In the GC code I've added "TODO do not init" here. Do you know why I might have added this? Its been a looong time...

Reply:
I was wondering the same myself I'm afraid :smile: Can't think what that would mean.. you definitely want to init the device haha

Reply:
I think my idea was to use `uint8_t command[3];`, since the initialization should not be required and takes time. However I am not sure, if an empty array is required. If not, we can remove the initialization. What do you think?

Reply:
Maybe you can try by setting it to `0xff, 0xff, 0xff` and check if the code still works.

Reply:
Will check this in a bit. Would've thought not initialising to 0 would result in invalid commands, because not all N64 commands are 3 bytes. In fact, the only ones that I use are 1-byte, so I imagine they could have junk data in them that causes the command to not be correctly recognised as 0x00 or 0x01. But maybe I'm wrong

Reply:
Okay, it does seem to work without the initialisation (or with 3x 0xFF), so I'll keep that initialisation out.

Reply:
Do we even need 3 bytes then? What if we use 1 byte?

Can you please also correct this for gamecube?

Reply:
Well, there are other commands which are 3 bytes. I just didn't implement those. They could be useful in future, although it's way more advanced

Reply:
Alright, thanks for the feedback! I think the PR is ready to merge then.

Reply:
Ok that's done and tested now 😃